nip47/event_handler_test.go

```diff
@@ -55,13 +55,13 @@ func TestCreateResponse(t *testing.T) {
 
 	nip47svc := NewNip47Service(svc.DB, svc.Cfg, svc.Keys, svc.EventPublisher)
 
-	res, err := nip47svc.CreateResponse(reqEvent, nip47Response, nostr.Tags{}, ss, svc.Keys.GetNostrSecretKey())
+	res, err := nip47svc.CreateResponse(reqEvent, nip47Response, nostr.Tags{}, nip47Cipher, svc.Keys.GetNostrSecretKey())
 	assert.NoError(t, err)
 	assert.Equal(t, reqPubkey, res.Tags.GetFirst([]string{"p"}).Value())
 	assert.Equal(t, reqEvent.ID, res.Tags.GetFirst([]string{"e"}).Value())
 	assert.Equal(t, svc.Keys.GetNostrPublicKey(), res.PubKey)
 
-	decrypted, err := nip04.Decrypt(res.Content, ss)
+	decrypted, err := nip47Cipher.Decrypt(res.Content)
```

Comment:
Here we only check 1.0 NIP-44 request event but not backwards compatibility with 0.0 NIP-04 request event, because we don't have access to the appWalletPrivKey here (although we can derive it here like we do in HandleEvent, but I feel it would look ugly then).

One solution I have in mind is to add a `LegacyEncrypt` and `LegacyDecrypt` methods to the cipher which can then help us decrypt it here using NIP-04

Reply:
I don't understand what you mean, we should have a separate test that creates a legacy request using NIP-04 that does not pass a version, and we should expect that one should use version 0.0 for the cipher

Reply:
Yes, but the way the tests are written is such that we first use `CreateAppWithPrivateKey` to get the app along with a shared secret (which is now replaced with a cipher), hence the cipher version is set to "1.0" for all.

I think an even better way is to pass the version too to `CreateAppWithPrivateKey` which would make sense as you have to specify the app's nwc version during it's creation. Will update this, thanks 🙌

Reply:
Cool. I think for legacy tests we should also support passing the version. This way we can test the 4 combinations.